pkg/clients/ec2/address.go

```diff
@@ -30,7 +31,8 @@ type AddressClient interface {
 
 // IsAddressNotFoundErr returns true if the error is because the address doesn't exist
 func IsAddressNotFoundErr(err error) bool {
-	if awsErr, ok := err.(smithy.APIError); ok {
+	var awsErr smithy.APIError
+	if errors.As(err, &awsErr) {
 		if awsErr.ErrorCode() == AddressAddressNotFound || awsErr.ErrorCode() == AddressAllocationNotFound {
```

Comment:
Have you validated that `errors.As` assign the actual struct into the provided pointer, i.e. `awsErr`? I remember looking at the source code of that function and it was doing only a type check.

Reply:
That is `errors.Is`, not `.As`?
https://pkg.go.dev/errors#pkg-functions
https://play.golang.org/p/CVmMXLy6_Bb